### PR TITLE
[W5.11][T09-A2]LIWEN LAI

### PR DESCRIPTION
**Pull request by Livian1107**
two enhancements have been updated:
1.add a new command, EditCommand, which can update the contact information of person in the addressbook.
2.persons in the addressbook are sorted with their name in alphabetical order.

### CODE REVIEW

**Review by chenxing1992 — COMMENTED** [review]
Codes are neat. Well done! Is there a better way to import specific util from person class?

File: doc/UserGuide.md [hunk 1]
```diff
@@ -66,6 +67,32 @@ Examples:
...
+Format: `edit INDEX NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...`
+
+> Using it's last displayed index from the address book, to update the information of the identified person
+> Words in `UPPER_CASE` are the parameters, items in `SQUARE_BRACKETS` are optional,
+> items with `...` after them can have multiple instances. Order of parameters are fixed.
+>
+> Put a `p` before the phone / email / address prefixes to mark it as `private`. `private` details can only
+> be seen using the `viewall` command.
+>
+> Persons can have any number of tags (including 0)
```
Comment (dalessr) [inline doc/UserGuide.md:81]: There is no need to explain those things not related to your new command

File: doc/UserGuide.md [hunk 2]
```diff
@@ -66,6 +67,32 @@ Examples:
...
+> Persons will sorted with name in alphabetical order after being added
+
+Examples:
+* `edit 1 John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
+* `edit 2 Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
+
+Examples:
```
Comment (dalessr) [inline doc/UserGuide.md:88]: Is this set of examples related to edit command?

File: src/seedu/addressbook/commands/AddCommand.java [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 import java.util.HashSet;
 import java.util.Set;
+import java.lang.*;
```
Comment (dalessr) [inline src/seedu/addressbook/commands/AddCommand.java:5]: This is a violation of coding styles. The exact class/interface need to be specified.

File: src/seedu/addressbook/commands/AddCommand.java [hunk 4]
```diff
@@ -15,7 +19,7 @@
 import seedu.addressbook.data.tag.UniqueTagList;
 
 /**
- * Adds a person to the address book.
+ * Adds a person to the address book with name in alphabetical order.
```
Comment (dalessr) [inline src/seedu/addressbook/commands/AddCommand.java:22]: The person list was originally sorted by chronological order. It might be easier for the users to find a person in the address book. In this case, is it better to create a sort command to let the user decide which order they prefer to list the persons?

File: src/seedu/addressbook/commands/EditCommand.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Edit the contact information(phone, email and address) of the certain person identified
```
Comment (dalessr) [inline src/seedu/addressbook/commands/EditCommand.java:13]: It is great to separate the long sentence into multiple lines :)

File: src/seedu/addressbook/data/AddressBook.java [hunk 6]
```diff
@@ -1,9 +1,6 @@
 package seedu.addressbook.data;
 
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (dalessr) [inline src/seedu/addressbook/data/AddressBook.java:3]: Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use sourcetree to look through changes before committing, and discard any unrelated changes.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 7]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:3]: Same issue as the above.

File: src/seedu/addressbook/parser/Parser.java [hunk 8]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (dalessr) [inline src/seedu/addressbook/parser/Parser.java:14]: Same issue as the above.

**Review by dalessr — COMMENTED** [review]
Good job on covering all aspects of the new command @Livian1107  👍 Please note that you need to create to PRs for the two distinct features :)

Some comments added. Please close the PR after reading comments.

File: doc/UserGuide.md (hunk 2 above)
Comment (Livian1107) [reply doc/UserGuide.md:88]: yes
